x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/IndexBasedTransformConfigManager.java

```diff
@@ -206,6 +211,43 @@ public void deleteOldTransformStoredDocuments(String transformId, ActionListener
         );
     }
 
+    @Override
+    public void deleteOldCheckpoints(String transformId, long checkpointLowerBound, long lowerBoundEpochMs, ActionListener<Long> listener) {
+        DeleteByQueryRequest deleteByQueryRequest = createDeleteByQueryRequest();
+        deleteByQueryRequest.indices(
+            TransformInternalIndexConstants.INDEX_NAME_PATTERN,
+            TransformInternalIndexConstants.INDEX_NAME_PATTERN_DEPRECATED
+        );
+        deleteByQueryRequest.setQuery(
+            QueryBuilders.boolQuery()
+                .filter(QueryBuilders.termQuery(TransformField.ID.getPreferredName(), transformId))
+                .filter(QueryBuilders.termQuery(TransformField.INDEX_DOC_TYPE.getPreferredName(), TransformCheckpoint.NAME))
+                .filter(QueryBuilders.rangeQuery(TransformCheckpoint.CHECKPOINT.getPreferredName()).lt(checkpointLowerBound))
```

Comment:
`checkpointLowerBound` signifies to me that it is the "lowest checkpoint bound" implying that that all checkpoints ABOVE this bound are to be deleted. 

I think `checkpointUpperBound` is better as everything less than this number is being deleted. 

Same observation for the timestamp filter.

Reply:
I changed it to: `deleteCheckpointsBelow` and `deleteOlderThan`, should be more descriptive. The java doc explains that _both_ must match.